src/primihub/algorithm/arithmetic.cc

```diff
@@ -227,13 +223,14 @@ int ArithmeticExecutor<Dbit>::execute() {
       } else
         mpc_op_exec_->MPC_Compare(sh_res);
       // reveal
-      if (party_id_ == 0) {
-        i64Matrix tmp;
-        tmp = mpc_op_exec_->reveal(sh_res);
-        for (size_t i = 0; i < tmp.rows(); i++)
-          cmp_res_.emplace_back(static_cast<bool>(tmp(i, 0)));
-      } else {
-        mpc_op_exec_->reveal(sh_res, 0);
+      for (auto party : parties_) {
```

Comment:
using const auto& instead of  auto to avoid copy

Reply:
OK, modification completed

Reply:
> using const auto& instead of auto to avoid copy

OK, modification completed